packages/snippet/src/UserInterface/Controller/Admin/SnippetAreaController.php

```diff
@@ -0,0 +1,129 @@
+use Sulu\Component\Rest\ListBuilder\Doctrine\DoctrineListBuilderFactoryInterface;
+use Sulu\Component\Rest\ListBuilder\Doctrine\FieldDescriptor\DoctrineFieldDescriptorInterface;
+use Sulu\Component\Rest\ListBuilder\Metadata\FieldDescriptorFactoryInterface;
+use Sulu\Component\Rest\RestHelperInterface;
+use Sulu\Messenger\Infrastructure\Symfony\Messenger\FlushMiddleware\EnableFlushStamp;
+use Sulu\Snippet\Application\Message\ModifySnippetAreaMessage;
+use Sulu\Snippet\Application\Message\RemoveSnippetAreaMessage;
+use Sulu\Snippet\Domain\Model\SnippetArea;
+use Sulu\Snippet\Domain\Model\SnippetAreaInterface;
+use Sulu\Snippet\Domain\Repository\SnippetAreaRepositoryInterface;
+use Sulu\Snippet\Infrastructure\Symfony\CompilerPass\SnippetAreaCompilerPass;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\HandleTrait;
+use Symfony\Component\Messenger\MessageBusInterface;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
+use Webmozart\Assert\Assert;
+
+/**
+ * @internal this class should not be instated by a project
+ *           Use instead a request or response listener to
+ *           extend the endpoints behaviours
+ *
+ * @phpstan-import-type Entry from SnippetAreaCompilerPass
+ */
+final class SnippetAreaController
+{
+    use HandleTrait;
+
+    /**
+     * @param array<Entry> $snippetArea
+     */
+    public function __construct(
+        MessageBusInterface $messageBus,
+        private NormalizerInterface $normalizer,
+        private FieldDescriptorFactoryInterface $fieldDescriptorFactory,
+        private DoctrineListBuilderFactoryInterface $listBuilderFactory,
+        private RestHelperInterface $restHelper,
+        private SnippetAreaRepositoryInterface $snippetAreaRepository,
+        private array $snippetArea,
+    ) {
+        // Setting the message bus of the HandleTrait
+        $this->messageBus = $messageBus;
+    }
+
+    public function cgetAction(Request $request): Response
+    {
+        /** @var DoctrineFieldDescriptorInterface[]|null $fieldDescriptors */
+        //$fieldDescriptors = $this->fieldDescriptorFactory->getFieldDescriptors(SnippetAreaInterface::RESOURCE_KEY);
+        //Assert::notNull($fieldDescriptors, 'Could not find field descriptors for resource key: ' . SnippetAreaInterface::RESOURCE_KEY);
+
+        /** @var DoctrineListBuilder $listBuilder */
+        //$listBuilder = $this->listBuilderFactory->create(SnippetAreaInterface::class);
+        //$listBuilder->setIdField($fieldDescriptors['id']); // We need to set this because it's the uuid doctrine column
+        //$listBuilder->setParameter('locale', $request->query->get('locale'));
+        //$listBuilder->setParameter('webspace', $request->query->get('webspace'));
+
+        //$this->restHelper->initializeListBuilder($listBuilder, $fieldDescriptors);
```

Comment:
// TODO: Currently I'm using the repository directly, as we need none of the functions of the list builder (pagination, filters, sorting etc)

Reply:
I would still use the listBuilder to have it more consistent and you never know, somebody may implement tons of snippet_area keys and therefore pagination may come in handy

Reply:
I agree, but as far as I know the frontend implementation doesn't support that. (And it wasn't present in 2.6)

But I couldn't get the join to work with the list XML.